source/kubernetes/minio-operator-reference.rst

```diff
@@ -104,94 +127,689 @@ The MinIO Operator adds a
+
+   *Optional*
+
+   The name of the 
+   :kube-docs:`Service Account 
+   <reference/access-authn-authz/service-accounts-admin/>` used to run all 
+   MinIO server :mc:`minio` pods created as part of the Tenant.
+
+.. kubeconf:: spec.subPath
+
+   *Optional*
+
+   The sub path appended to the :kubeconf:`spec.mountPath`. The resulting
+   full path is the directory in which MinIO stores data.
+
+   For example, given a :kubeconf:`~spec.mountPath` of ``export`` and
+   a :kubeconf:`~spec.subPath` of ``minio``, the full mount path is
+   ``export/minio``.
+
+   Defaults to empty (``""``).
+
+.. kubeconf:: spec.zones
+
+   *Required*
+
+   The configuration for each MinIO Zone deployed in the MinIO Tenant.
+
+
+
+
+     
+
+Transport Layer Encryption (TLS)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. parsed-literal::
+
+   spec:
+      :kubeconf:`~spec.requestAutoCert`: <boolean>
+      :kubeconf:`~spec.certConfig`:
+         :kubeconf:`~spec.certConfig.commonName`: <string>
+         :kubeconf:`~spec.certConfig.dnsNames`: <string>
+         :kubeconf:`~spec.certConfig.organizationName`: <string>
+         :kubeconf:`~spec.externalCertSecret`: 
+            - name: <string>
+              type: kubernetes.io/tls
+         :kubeconf:`~spec.externalClientCertSecret`:
+            name: <string>
+            type: kubernetes.io/tls
+
+.. kubeconf:: spec.requestAutoCert
+
+   *Optional*
+
+   Specify ``true`` to enable automatic TLS certificate generation and
+   signing using the Kubernetes ``certificates.k8s.io`` API. See the
+   Kubernetes documentation on
+   :kube-docs:`Manage TLS Certificates in a Cluster 
+   <tasks/tls/managing-tls-in-a-cluster/>` for more information.
+
+.. kubeconf:: spec.certConfig
+
+   *Optional*
+
+   The configuration settings to use when auto-generating x.509 certificates for
+   TLS encryption.
+
+   Omit to use auto-generated self-signed certificates.
+
+.. kubeconf:: spec.certConfig.commonName
+
+   *Optional*
+
+   The x.509 Common Name to use when generating x.509 certificates for TLS
+   encryption.
+
+.. kubeconf:: spec.certConfig.dnsNames
+
+   *Optional*
+
+   The DNS names to use when generating x.509 certificates for TLS encryption.
+
+.. kubeconf:: spec.certConfig.organizationName
+
+   *Optional*
+
+   The x.509 Organization Name to use when generating x.509 certificates for
+   TLS encryption.
+
+.. kubeconf:: spec.externalCertSecret
+
+   *Optional*
+
+   One or more Kubernetes secrets that contain custom TLS certificate and 
+   private key pairs for enabling TLS with SNI support on the MinIO Tenant
+   pods.
+
+   Each item in the array contains an object where:
+
+   - ``names`` specifies the name of the Kubernetes secret, and
+   - ``types`` specifies ``kubernetes.io/tls``
+
+   .. code-block:: yaml
+
+      spec:
+         externalCertSecret:
+            - name: tenant-external-cert-secret-name
+              type: kubernetes.io/tls
+
+.. kubeconf:: spec.externalClientCertSecret
```

Comment:
@nitisht Is there a reason to not also have an array here vs an object? A MinIO Tenant could have clients using multiple custom CAs right?

Reply:
No reason as such. May be we can make it an array going forward.